tests/functional/strategy/test_misc.py

```diff
@@ -134,3 +134,55 @@ def test_set_metadataURI(gov, strategy, rando):
     assert strategy.metadataURI() == "ipfs://test2"
     with brownie.reverts():
         strategy.setMetadataURI("ipfs://fake", {"from": rando})
+
+
+def test_sandwich_attack(
+    chain, TestStrategy, web3, token, gov, vault, strategist, rando
+):
+
+    honest_lp = gov
+    attacker = rando
+    balance = token.balanceOf(honest_lp) / 2
+
+    # seed attacker their funds
+    token.transfer(attacker, balance, {"from": honest_lp})
+
+    # we don't use the one in conftest because we want no rate limit
+    strategy = strategist.deploy(TestStrategy, vault)
+    vault.setManagementFee(0, {"from": gov})
+    vault.setPerformanceFee(0, {"from": gov})
+    vault.addStrategy(strategy, 4_000, 0, 2 ** 256 - 1, 0, {"from": gov})
+    vault.updateStrategyPerformanceFee(strategy, 0, {"from": gov})
+
+    strategy.harvest({"from": strategist})
+    # strategy is returning 0.02%. Equivalent to 35.6% a year at 5 harvests a day
+    profit_to_be_returned = token.balanceOf(strategy) / 5000
+    token.transfer(strategy, profit_to_be_returned, {"from": honest_lp})
+
+    # now for the attack
+
+    # attacker sees harvest enter tx pool
+    attack_amount = balance
+
+    print(f"Attack Capital: {attack_amount}")
```

Comment:
No print statements please, use `assert`

Reply:
done in latest